app/code/Magento/Swatches/Model/SwatchAttributesProvider.php

```diff
@@ -17,6 +18,10 @@
  */
 class SwatchAttributesProvider
 {
+    /**
+     * @var \Magento\Swatches\Helper\Data
+     */
+    protected $swatchesHelper;
```

Comment:
Looks like this dependency is redundant, please, remove it, and remove usage as constructor argument